Make contact pressure a generic variable for all contact pairs

recuero opened the pull request:
If names are not distinct, these will be overridden and only one of the contact pairs will have the correct contact pressure or penetration information. This will require regolding tests due to the name change. 

 (#18022)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1255357/) on a6a18c4 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22886/site)

This comment will be updated on new commits.

recuero committed:
Create single contact pressure variable for multiple contact pair setup (#18022)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1283551/) on c3a5f5d wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/22886/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/22886/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format c81937639f08fe075fd1b17d6514c530e701e8e0`

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1284565/) on b6998ac wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>6095ff</th>
      <th colspan=4>#22886 b6998a</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.53%</td>
      <td>84.53%</td>
      <td>-0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>81635</td>
      <td>81630</td>
      <td>-5</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14936</td>
      <td>14941</td>
      <td>+5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22886/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22886/coverage/framework)


## Modules coverage

### Contact

<table>
  <thead>
    <tr>
      <th></th>
      <th>6095ff</th>
      <th colspan=4>#22886 b6998a</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.96%</td>
      <td>89.97%</td>
      <td>+0.01%</td>
      <td>97.44%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3315</td>
      <td>3327</td>
      <td>+12</td>
      <td>38</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>370</td>
      <td>371</td>
      <td>+1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/22886/coverage_diff/contact)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/22886/coverage/contact)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/22886/coverage/xfem)

</details>



This comment will be updated on new commits.

recuero committed:
Generate single contact pressure output for multiple contact action objects

Also change the nodal_area name to be just "nodal_area" and remove the
suffix given by the action name.

recuero committed:
Regolding of contact and combined module

Changes only reflect the fact that the nodal area variable created by
the contact action now does not have a suffix with the name of the
contact action object.

For example, before we would have nodal_area_action_1 and
nodal_area_action_2 for contact action objects named action_1 and
action_2. Now, the only variable name would be nodal_area.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1285606/) on 095f5f9 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/22886/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/22886/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format cebfc510d280dbd6eee799bb1cbf857b6b6a2c2a`

recuero commented:
This PR requires downstream updates.

bwspenc commented on modules/contact/test/tests/multiple_contact_pairs/tests:
```diff
@@ -1,13 +1,36 @@
     rel_err = 1.0e-5
     cli_args = "Postprocessors/active=''"
-    requirement = 'Contact module action shall allow for multiple contact pairs when selecting node-face formulation.'
+    requirement = 'Contact module action shall allow for multiple contact pairs when selecting '
```
`'The system shall set up node-face interactions between multiple surface pairs using the Contact action'`

recuero committed:
Address reviewer's comments

recuero commented:
Comments should have been addressed.

bwspenc reviewed (commented):
All my comments have been addressed. I'm ready to approve this, but I'll hold off until we get the OK to break next.

recuero commented:
```
At least 2 approving reviews are required to merge this pull request.
```
